go/integration_test/processor/calls_test.go

```diff
@@ -193,20 +186,266 @@ func TestProcessor_DifferentCallTypesAccessStorage(t *testing.T) {
 				if err != nil || !result.Success {
 					t.Errorf("execution was not successful or failed with error %v", err)
 				}
-				if test.sameStorage {
+				if call.sameStorage {
 					if !slices.Equal(result.Output, append(bytes.Repeat([]byte{0}, 31), byte(42))) {
-						t.Errorf("%s did not access the same storage, got unexpected output: %v", test.call.String(), result.Output)
+						t.Errorf("%s did not access the same storage, got unexpected output: %v", callName, result.Output)
 					}
 				} else {
 					if !slices.Equal(result.Output, bytes.Repeat([]byte{0}, 32)) {
-						t.Errorf("%s did access the same storage, got unexpected output: %v", test.call.String(), result.Output)
+						t.Errorf("%s did access the same storage, got unexpected output: %v", callName, result.Output)
 					}
 				}
 			})
 		}
 	}
 }
 
+func TestProcessor_DifferentCallTypesHandleValueCorrectly(t *testing.T) {
+	gasLimit := tosca.Gas(1000000)
+	for processorName, processor := range getProcessors() {
+		if strings.Contains(processorName, "floria") {
+			continue // todo implement different call types
+		}
+		for callName, call := range callTypesAndProperties() {
+			t.Run(fmt.Sprintf("%s-%s", processorName, callName), func(t *testing.T) {
+				sender0 := tosca.Address{1}
+				receiver0 := tosca.Address{2}
+				receiver1 := tosca.Address{3}
+				senderBalance := tosca.NewValue(100)
+				transferValue := tosca.NewValue(42)
+
+				// set call arguments
+				code0 := pushCallArguments(call, gasLimit, tosca.Value{}, receiver1)
```

Comment:
An idea for a small improvement: you could set a different value here to also check for those call types that do have a value parameter that the value passed as an argument is reaching the inner call